src/celeritas/optical/action/ActionLauncher.device.hh

```diff
@@ -44,15 +44,15 @@ namespace optical
  }
  * \endcode
  */
-template<class F>
+template<class F, class StepActionT = OpticalStepActionInterface>
 class ActionLauncher : public KernelLauncher<F>
 {
     static_assert(
         (std::is_trivially_copyable_v<F> || CELERITAS_USE_HIP
          || CELER_COMPILER == CELER_COMPILER_CLANG)
             && !std::is_pointer_v<F> && !std::is_reference_v<F>,
         R"(Launched action must be a trivially copyable function object)");
-    using StepActionT = OpticalStepActionInterface;
+    // using StepActionT = OpticalStepActionInterface;
```

Comment:
I think only the constructors need to be templated on the step action. It's purely for obtaining the label for diagnostic export.